pkg/proxy/topology.go

```diff
@@ -160,19 +164,23 @@ func canUseTopology(endpoints []Endpoint, svcInfo ServicePort, nodeLabels map[st
 		}
 	}
 
-	zone, ok := nodeLabels[v1.LabelTopologyZone]
-	if !ok || zone == "" {
-		klog.V(2).InfoS("Skipping topology aware endpoint filtering since node is missing label", "label", v1.LabelTopologyZone)
-		return false
-	}
-
+	zone, foundZone := nodeLabels[v1.LabelTopologyZone]
 	hasEndpointForZone := false
 	for _, endpoint := range endpoints {
 		if !endpoint.IsReady() {
 			continue
 		}
+
+		// If any of the endpoints do not have zone hints, we bail out
 		if endpoint.ZoneHints().Len() == 0 {
-			klog.V(2).InfoS("Skipping topology aware endpoint filtering since one or more endpoints is missing a zone hint", "endpoint", endpoint)
+			return false
+		}
+
+		// If there is at least one zone hint, we check if there is a zone label, if there isn't one we bail out
+		if !foundZone || zone == "" {
+			logOnce.Do(func() {
+				klog.V(4).InfoS("Skipping topology aware endpoint filtering since node is missing label", "label", v1.LabelTopologyZone)
```

Comment:
Now that we've dramatically limited the number of cases this could trigger, I think I'd prefer keeping 2 as the log level since we're now quite confident that something is not working as intended.

Reply:
ok